### ISSUE TITLE
Badge: Icon Only variant in storybook does not have aria-label.

### ISSUE THREAD

**Issue by dineshy87** [adobe/react-spectrum#4043]
# 🙋 Documentation Request

According to [React-Spectrum Docs](https://react-spectrum.adobe.com/react-spectrum/Badge.html#accessibility), if a visible label isn't specified, an aria-label must be provided for accessibility. But there are no examples provided.

[Storybook example](https://reactspectrum.blob.core.windows.net/reactspectrum/41e26fa0f01ee9ab6e5c21edf6f7bb5d498bf40f/storybook/index.html?path=/story/badge--icon-only&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom) doesn’t have any aria-label and so it’s completely skipped by screenreader users.

[Spectrum guidance](https://spectrum.adobe.com/page/badge/#Avoid-using-icon-only-badges) says that Icon-only variants should be avoided, but if used, should always include a tooltip on hover.

**Comment by snowystinger** [comment]
> According to [React-Spectrum Docs](https://react-spectrum.adobe.com/react-spectrum/Badge.html#accessibility), if a visible label isn't specified, an aria-label must be provided for accessibility. But there are no examples provided.

Icon only should be avoided, we don't want to encourage that by providing an example of it. Instead, we let people know they need to provide an aria-label if they decide to make an icon only badge. This is a common statement across our documentation.

> [Storybook example](https://reactspectrum.blob.core.windows.net/reactspectrum/41e26fa0f01ee9ab6e5c21edf6f7bb5d498bf40f/storybook/index.html?path=/story/badge--icon-only&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom) doesn’t have any aria-label and so it’s completely skipped by screenreader users.

We should add it for completeness, but it's not documentation, only for internal use.

> [Spectrum guidance](https://spectrum.adobe.com/page/badge/#Avoid-using-icon-only-badges) says that Icon-only variants should be avoided, but if used, should always include a tooltip on hover.

Spectrum guidance is a bit behind on this one I believe. Badges are not focusable, so they currently can't have a tooltip.
